blog_migration/importBlogPosts.js

```diff
@@ -0,0 +1,117 @@
+    // The post body contains some weird non-standard <link> tags that
+    // look like this:
+    //
+    // <link https://example.com/path/to/somewhere>A document</link>
+    // <link alice@example.com - mail>alice[at]example.com</link>
+    // <link //domain/path/to/file.pdf>A document</link>
+    // <link alice@example.com - mail>alice[at]example.com</link>
+    //
+    // The following line converts these to the [...](...) Markdown syntax.
+    //
+    // Many posts also contain links to internal pages of the old site,
+    // referring to them by their internal page ID like this:
+    //
+    // <link 57 - internal link>...</link>
+    //
+    // Most of these we don't have any equivalent for, so we map them
+    // to the (non-existent) /blog/internal/:id for now...
+    .replace(/<link (\d+) .*?>(.*?)<\/link>/g, '[$2](/blog/internal/$1)')
+    .replace(/<link (\S+) - mail.*?>(.*?)<\/link>/g, '[$2](mailto:$1)')
+    .replace(/<link (\S+?)>(.*?)<\/link>/g, '[$2]($1)')
+    .replace(/<link (\S+).*?>(.*?)<\/link>/g, '[$2]($1)')
+    // Some posts' content starts with blank lines that we don't want to keep.
+    .trim()
+}
+
+function createAssetLink(assetUrl) {
+  // The asset URLs look like this:
+  // //images.contentful.com/6jocdllnp50q/4C5fiodkVWCCaka8cUCqUy/...
+  // The second path segment is the asset ID.
+  const assetId = assetUrl.split('/')[4]
+
+  return {
+    sys: {
+      type: 'Link',
+      linkType: 'Asset',
+      id: assetId,
+    },
+  }
+}
+
+async function importBlogPost(post, space) {
+  const fields = {
+    title: { de: post.title },
+    slug: { de: slug(post.title, { lower: true }) },
+    metaDescription: { de: post.teaser },
+    teaserImage: { en: createAssetLink(post.teaserImage) },
+    date: { en: new Date(post.datetime * 1000).toISOString() },
+    content: { de: preparePostContent(post.bodytext) },
+  }
+
+  const id = `imported-blog-post-${post.uid}`
+  let entry = await space.getEntry(id).catch(() => null)
+
+  if (entry) {
+    Object.assign(entry.fields, fields)
+    entry = await entry.update()
+  } else {
+    entry = await space.createEntryWithId('blogPost', id, { fields })
+  }
+
+  await entry.publish()
+}
+
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+async function main() {
+  const client = contentfulManagement.createClient({
+    accessToken: env.CONTENTFUL_MANAGEMENT_ACCESS_TOKEN,
+  })
+  const space = await client.getSpace(env.CONTENTFUL_SPACE_ID)
+  const posts = loadPostsFromExport()
+
+  // To not hit the Contentful API rate limit, we need to import the
+  // posts one by one with some delay between requests.
+
+  /* eslint-disable */
+  for (const post of posts) {
+    console.log(`Importing post ${post.uid}…`)
+    await importBlogPost(post, space)
+    await delay(50)
+  }
+  /* eslint-enable */
+
+  // eslint-disable-next-line no-console
+  console.log('Done!')
+}
+
+// eslint-disable-next-line no-console
```

Comment:
You could probably disable Eslint for all file 😄

Reply:
Yeah, I'm going to move a few `eslint-disable` things to the top of the file. I want to preserve as much ESLint assistance as possible, though.